docs/api/json.md

```diff
@@ -0,0 +1,291 @@
+
+### The raw format
+
+This format is applicable to append-only event streams where rows can only
+be inserted but not deleted.  In this case, a data change event can be
+represented as a SQL record without any additional framing, with the insert
+operation being implied.
+
+### The insert/delete format
+
+A data change event in this format is represented as a JSON object with a
+single key, which must be equal to `insert` or `delete`.  The associated value
+represents a table row to be inserted or deleted, encoded using the format
+documented above, e.g.:
+
+```json
+{"delete": {"part": 1, "vendor": 2, "price": 10000}}
+{"insert": {"part": 1, "vendor": 2, "price": 30000}}
+```
+
+## Encoding multiple changes
+
+Data change events are exchanged as data streams transmitted over transports such
+as HTTP or Kafka.  We use newline-delimited JSON (NDJSON) to assemble individual
+data change events into a stream.  In NDJSON, each line within the data stream
+represents a self-contained and valid JSON document, in this case, a data change
+event:
+
+```json
+{"delete": {"part": 1, "vendor": 2, "price": 10000}}
+{"insert": {"part": 1, "vendor": 2, "price": 30000}}
+{"insert": {"part": 2, "vendor": 3, "price": 34000}}
+...
+```
+
+A JSON event stream can be [configured](#configuring-json-event-streams)
+to combine multiple events into JSON arrays.  This format still uses NDJSON,
+where each line contains an array of data change events:
+
+```json
+[{"delete": {"part": 1, "vendor": 2, "price": 10000}}, {"insert": {"part": 1, "vendor": 2, "price": 30000}}]
+[{"insert": {"part": 2, "vendor": 3, "price": 34000}}, {"delete": {"part": 3, "vendor": 1, "price": 5000}}]
+...
+```
+
+This format allows one to break up the event stream into transport
+messages so that each message contains a valid JSON document
+by encoding all events in the message as an array.
+
+## Configuring JSON event streams
+
+### Configure connectors via the Feldera Web Console
+
+When creating a new input or output connector using the Feldera Web Console,
+the data format is configured in the `FORMAT` section of the connector
+creation wizard:
+
+1. choose `JSON` from the list of supported formats
+1. choose appropriate data change event encoding (`Raw` or `Insert & Delete`)
+1. enable array encapsulation if the stream contains events grouped in arrays
+
+![Data change event format selector](format_selector.png)
+
+See also the [input/output connector tutorial](tutorials/basics/part3.md).
+
+### Configure connectors via the REST API
+
+When creating or modifying connectors via the REST API
+[`/connectors` endpoint](/api/create-a-new-connector),
+the data format is specified in the `format` field of
+the connector configuration:
+
+```json
+{
+    "transport": {
+        "name": "url",
+        "config": {
+            "path":"https://feldera-basics-tutorial.s3.amazonaws.com/part.json"
+        }
+    },
+    "format": {
+        // Choose JSON format for the connector.
```

Comment:
oh // is also a linter error in json, nevermind my comments above then

Reply:
There the json5 format which accepts comments.